tests/test-sources/modules/performance/byte-compile-lua.nix

```diff
@@ -325,70 +327,62 @@ in
-          (setDeps pkgs.vimPlugins.lz-n [ say ])
-          # 'nvim-cmp' depends on 'argparse'
-          (setDeps pkgs.vimPlugins.nvim-cmp [ lua.pkgs.argparse ])
-        ];
+      extraPlugins = pluginStubs.pluginPack;
 
       extraConfigLuaPost = ''
+        ${pluginStubs.pluginChecks}
+
         ${isByteCompiledFun}
 
-        -- Plugins themselves are importable
-        require("lz.n")
-        require("cmp")
+        -- Lua modules are byte-compiled
+        ${lib.concatMapStringsSep "\n" (
+          name: "test_lualib_file(require('${name}').name, true)"
+        ) pluginStubs.libNames}
 
-        -- Lua modules are importable and byte compiled
-        local say = require("say")
-        test_lualib_file(say.set, true)
-        local argparse = require("argparse")
-        test_lualib_file(argparse("test").parse, true)
+        -- Plugins themselves are not byte-compiled
+        ${lib.concatMapStringsSep "\n" (
+          name: "test_rtp_file('lua/${name}/init.lua', false)"
+        ) pluginStubs.pluginNames}
       '';
 
       assertions =
         let
-          requiredLuaModules = lib.pipe config.extraPlugins [
-            (builtins.catAttrs "requiredLuaModules")
-            builtins.concatLists
+          # Get plugins with all dependencies
+          getDeps = drv: [ drv ] ++ builtins.concatMap getDeps drv.dependencies or [ ];
+          plugins = lib.pipe config.build.plugins [
+            (builtins.catAttrs "plugin")
+            (builtins.concatMap getDeps)
+            lib.unique
+          ];
+          getPropagatedBuildInputs = lib.flip lib.pipe [
+            (
+              drvs: builtins.catAttrs "propagatedBuildInputs" drvs ++ builtins.catAttrs "requiredLuaModules" drvs
+            )
+            (builtins.concatMap (deps: deps ++ getPropagatedBuildInputs deps))
             lib.unique
           ];
+          propagatedBuildInputs = getPropagatedBuildInputs plugins;
         in
         [
-          # Ensure that propagatedBuildInputs are byte-compiled recursively
-          # by checking that every library is present only once
+          # Ensure that lua dependencies are byte-compiled recursively
+          # by checking that every library is present only once.
+          # If there are two different derivations with the same name,
+          # then one of them probably isn't byte-compiled.
           {
-            assertion = lib.allUnique (map lib.getName requiredLuaModules);
+            assertion = lib.allUnique (map lib.getName propagatedBuildInputs);
             message = ''
-              Expected requiredLuaModules of all propagatedBuildInputs to have unique names.
-              Got the following derivations: ${builtins.concatStringsSep ", " requiredLuaModules}.
-              One possible reason is that not all dependencies are overridden the same way.
+              Expected propagatedBuildInputs and requiredLuaModules of all plugins to have unique names.
```

Comment:
Message mentions _both_ `propagatedBuildInputs` _and_ `requiredLuaModules`, but the assertion only checks one. Is that correct?

EDIT: my mistake, I see they're concatenated above. Maybe the name should reflect that? E.g. `compiledLuaInputs` or `compiledLuaModules` or `allInputs` or something?